### DISCUSSION TITLE
How to use obsidian with excalidraw plus?

### DISCUSSION THREAD

**Discussion by dnson** [zsviczian/obsidian-excalidraw-plugin#588] [General]
I do not know how to login with excalidraw plus. Could you please help?

**Comment by zsviczian** [comment]
I am sorry, but Obsidian Excalidraw does not support Excalidraw plus at the moment.

**Reply by mtin79** [reply]
Any plans, workarounds or roadmap to support Excalidraw plus?

**Reply by zsviczian** [reply]
Which feature of Excalidraw Plus are you missing?

"Never say never", however I don't think Excalidraw plus support will ever be offered in the Obsidian plugin... In an nutshell there are couple of issue that I don't see how I could overcome:
- Obsidian is a local first note taking system. Everything is stored in your vault. This means I store images and the drawing locally. Synchronizing local images with items from the cloud opens up a whole host of issues that impact the security and integrity of your entire Vault.
- The Obsidian plugin has lots of custom features such as local markdown link support, text transclusions, etc. These are not implemented in excalidraw.com and also I don't quite see how I'd convert a link when synchronizing to the cloud.

If your need is to collaborate with others occasionally, then you can do that on Excalidraw.com. If you save the collaboration room id, then you can embed that drawing to your Obsidian document in an iFrame, or you can CTRL+A, CTRL+C, CTRL+V the entire drawing into your local copy in Obsidian.

You can even automate some of this process with Templater. Here's a script that creates a new collaboration room and inserts an iFrame:

```javascript
<%
cont getExcalidrawIFrame = () => { 
  const room = Array.from(window.crypto.getRandomValues(new Uint8Array(10))).map((byte) => `0${byte.toString(16)}`.slice(-2)).join("");
  const key = (await window.crypto.subtle.exportKey("jwk",await window.crypto.subtle.generateKey({name:"AES-GCM",length:128},true,["encrypt", "decrypt"]))).k;
  return `\n\n<iframe src="https://excalidraw.com/#room=${room},${key}`" style="width:800px;height:500px;resize:both;overflow:auto;"/>\n\n`
}

tR += getExcalidrawIFrame();
%>
```

**Reply by mtin79** [reply]
> Which feature of Excalidraw Plus are you missing?
> 
> 
> 
> "Never say never", however I don't think Excalidraw plus support will ever be offered in the Obsidian plugin... In an nutshell there are couple of issue that I don't see how I could overcome:
> 
> - Obsidian is a local first note taking system. Everything is stored in your vault. This means I store images and the drawing locally. Synchronizing local images with items from the cloud opens up a whole host of issues that impact the security and integrity of your entire Vault.
> 
> - The Obsidian plugin has lots of custom features such as local markdown link support, text transclusions, etc. These are not implemented in excalidraw.com and also I don't quite see how I'd convert a link when synchronizing to the cloud.
> 
> 
> 
> If your need is to collaborate with others occasionally, then you can do that on Excalidraw.com. If you save the collaboration room id, then you can embed that drawing to your Obsidian document in an iFrame, or you can CTRL+A, CTRL+C, CTRL+V the entire drawing into your local copy in Obsidian.
> 
> 
> 
> You can even automate some of this process with Templater. Here's a script that creates a new collaboration room and inserts an iFrame:
> 
> 
> 
> ```javascript
> 
> <%
> 
> cont getExcalidrawIFrame = () => { 
> 
>   const room = Array.from(window.crypto.getRandomValues(new Uint8Array(10))).map((byte) => `0${byte.toString(16)}`.slice(-2)).join("");
> 
>   const key = (await window.crypto.subtle.exportKey("jwk",await window.crypto.subtle.generateKey({name:"AES-GCM",length:128},true,["encrypt", "decrypt"]))).k;
> 
>   return `\n\n<iframe src="https://excalidraw.com/#room=${room},${key}`" style="width:800px;height:500px;resize:both;overflow:auto;"/>\n\n`
> 
> }
> 
> 
> 
> tR += getExcalidrawIFrame();
> 
> %>
> 
> ```

Thanks for the detailed reply.
Exactly the last bit is the most important for me but as you say it's rather an issue to have a shortcut : template to automatically create an excalidraw plus drawing that is connected to the account and manageable/shareable via the plus account.
So initially I was looking for an api setting or login credentials in the plugin that would create and use excalidraw connected to the plus account.
Maybe I didn't understand fully how each on could integrate.

Cheers!
